iq/iq.go

```diff
@@ -0,0 +1,221 @@
+		warnUserOfBadLifeChoices()
+	}
+
+	internalID := getInternalApplicationID(applicationID)
+	newPurls := getPurls(purls)
+	sbom := cyclonedx.ProcessPurlsIntoSBOM(newPurls)
+	statusURL := submitToThirdPartyAPI(sbom, internalID)
+
+	finished := make(chan bool)
+
+	statusURLResp = StatusURLResult{}
+
+	go func() {
+		for {
+			select {
+			case <-finished:
+				return
+			default:
+				pollIQServer(fmt.Sprintf("%s/%s", localConfig.Server, statusURL), finished)
+				time.Sleep(pollInterval)
+			}
+		}
+	}()
+
+	<-finished
+	return statusURLResp
+}
+
+func getInternalApplicationID(applicationID string) (internalID string) {
+	client := &http.Client{}
+
+	req, err := http.NewRequest(
+		"GET",
+		fmt.Sprintf("%s%s%s", localConfig.Server, internalApplicationIDURL, applicationID),
+		nil,
+	)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		var response applicationResponse
+		json.Unmarshal(bodyBytes, &response)
+		return response.Applications[0].ID
+	}
+	return ""
+}
+
+func submitToThirdPartyAPI(sbom string, internalID string) string {
+	client := &http.Client{}
+
+	url := fmt.Sprintf("%s%s", localConfig.Server, fmt.Sprintf("%s%s%s%s", thirdPartyAPILeft, internalID, thirdPartyAPIRight, localConfig.Stage))
+
+	req, err := http.NewRequest(
+		"POST",
+		url,
+		bytes.NewBuffer([]byte(sbom)),
+	)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	req.Header.Set("Content-Type", "application/xml")
+	req.Header.Set("User-Agent", fmt.Sprintf("nancy-client/%s", buildversion.BuildVersion))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusAccepted {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		var response thirdPartyAPIResult
+		json.Unmarshal(bodyBytes, &response)
+		return response.StatusURL
+	}
+
+	return ""
+}
+
+func pollIQServer(statusURL string, finished chan bool) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", statusURL, nil)
+
+	req.SetBasicAuth(localConfig.User, localConfig.Token)
+
+	req.Header.Set("User-Agent", fmt.Sprintf("nancy-client/%s", buildversion.BuildVersion))
+
+	resp, err := client.Do(req)
+
+	if err != nil {
+		finished <- true
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			fmt.Print(err)
+		}
+		var response StatusURLResult
+		json.Unmarshal(bodyBytes, &response)
+		statusURLResp = response
+		if response.IsError {
+			finished <- true
+		}
+		finished <- true
+	}
+	fmt.Print(".")
+}
+
+func warnUserOfBadLifeChoices() {
```

Comment:
![best](https://user-images.githubusercontent.com/1099906/71704067-1c400480-2da6-11ea-9a93-8ee8da875015.gif)
#bestMethodName

Reply:
LOL I knew someone would like that

Reply:
beautiful